src/server/active_tran_server.cpp

```diff
@@ -164,8 +161,30 @@ active_tran_server::push_request (ats_to_ps_request reqid, std::string &&payload
   m_ps_request_queue->push (reqid, std::move (payload));
 }
 
-void
-active_tran_server::receive_saved_lsa (cubpacking::unpacker &upk)
+void active_tran_server::receive_log_page (cubpacking::unpacker &upk)
+{
+  std::string message;
+  upk.unpack_string (message);
+
+  int error_code;
+  memcpy (&error_code, message.c_str (), sizeof (error_code));
+
+  LOG_PAGEID pageid = 0;
+  if (error_code == NO_ERROR)
+    {
+      LOG_PAGE *logpage = new LOG_PAGE ();
+      memcpy (& (logpage->hdr), message.c_str () + sizeof (error_code), sizeof (logpage->hdr));

```

Comment:
An entire log page should be transferred.
```suggestion
      char log_page_buffer[IO_MAX_PAGE_SIZE];
      std::memcpy (log_page_buffer, message.c_str () + sizeof (error_code), db_log_page_size ());

```

Reply:
That's how I initially wrote the code and I had a crash on the tran server side. Let me see what I can do about it..